webapp/store/snap_details_views.py

```diff
@@ -237,6 +240,20 @@ def snap_details(snap_name):
             os_metrics = None
             country_devices = None
 
+        if authentication.is_authenticated(flask.session):
+            try:
+                account_info = publisher_api.get_account(flask.session)
+            except ApiError as api_error:
+                # TODO: handle errors including macaroon refresh
```

Comment:
@tbille @jpmartinspt 

Is it OK for us to be calling publisher API and logic from snap details view? We want to get list of users snaps (if they are signed in) to know if they can edit current snap.
Also, calling publisher API would require different error handling then snap details views, at least for macaroon refresh redirect to work...

Can it be done here? Or should we do it differently?

Reply:
mmm I am not sure this is the best way to do this. Let's chat @jpmartinspt if we could do something better.

Reply:
@tbille Other idea we discussed with Luke was to store list of user snaps in session when the log in. But not sure what other implications this may have.

Reply:
this was this idea that I had. Since this is not something sensitive.

Reply:
I think I like the session idea too.

Reply:
@jpmartinspt I guess as long as we use it just for notification it will not be as bad, probably may be hardly noticeable.
We may consider refreshing the data in session when visiting "my snaps" page, so at least no logout will be required.